TwitchDownloaderCore/Extensions/M3U8Extensions.cs

```diff
@@ -118,5 +118,14 @@ public static string GetResolutionFramerateString(this M3U8.Stream stream)
 
             return $"{frameHeight}p{frameRate}";
         }
+
+        /// <summary>
+        /// Returns the best quality stream from the provided M3U8.
+        /// </summary>
+        public static M3U8.Stream BestQualityStream(this M3U8 m3u8)
+        {
+            var source = Array.Find(m3u8.Streams, x => x.MediaInfo.Name.Equals("source", StringComparison.OrdinalIgnoreCase));
```

Comment:
Sometimes "source" is called "chunked".

Reply:
`MediaInfo.Name` can be `chunked`? I queried this a ton today while testing and never saw that occur. `StreamInfo.Video` is always `chunked`, but I never saw a value for `MediaInfo.Name` other than `1080p60` or `Source`.

Reply:
Correction: this should be `MediaInfo.Name.Contains("source", StringComparison.OrdinalIgnoreCase) || MediaInfo.GroupID.Equals("chunked", StringComparison.OrdinalIgnoreCase)`.

See https://github.com/lay295/TwitchDownloader/blob/f166581dfdecdb020b441cda5efc02c5f9a432c2/TwitchDownloaderCore.Tests/M3U8ExtensionTests.cs for an example M3U8

Reply:
Oh I didn't see that test file, I'll add tests for this new extension as well.

However, I don't see any examples in there where `MediaInfo.Name` ends up as "chunked". On line 29, "chunked" is the `groupId`, and `Name` ends up as `1080p60 (source)`. I can add a test where the name is "chunked" if it needs to exist, but I don't see that currently being tested.

Reply:
> MediaInfo.Name

Yeah, I meant to say `MediaInfo.GroupId`, not `Name`

Reply:
Just added in e90a07d707d3a011e16bfc4ee71db666134c4128.